services/construction/types.go

```diff
@@ -181,4 +182,10 @@ type Client interface {
 	ParseOps(
 		tx *evmClient.LoadedTransaction,
 	) ([]*RosettaTypes.Operation, error)
+
+	// EstimateGas tries to estimate the gas needed to execute a specific transaction based on
+	// the current pending state of the backend blockchain. There is no guarantee that this is
+	// the true gas limit requirement as other transactions may be added or removed by miners,
+	// but it should provide a basis for setting a reasonable default.
+	EstimateGas(ctx context.Context, msg Eth.CallMsg) (uint64, error)
```

Comment:
`EstimateGas` is the built-in function of ethclient, it shouldn't be in our client interface